### ISSUE TITLE
Inconsistency between lisp and sectorlisp.bin quote

### ISSUE THREAD

**Issue by WlarusFromDaSpace** [jart/sectorlisp#11]
The program "lisp" behaves differently from sectorlisp.bin which, I believe, is unintended.
Output of lisp:
```
THE LISP CHALLENGE V1
VISIT GITHUB.COM/JART
(QUOTE (1 1))
(1)
(QUOTE (1 1))
(1 1∙NIL)
```
Output of sectorlisp.bin when run with `qemu-system-x86_64 -nographic -fda sectorlisp.bin`:
```
SeaBIOS (version d55cb5a)


iPXE (http://ipxe.org) 00:03.0 CA00 PCI2.10 PnP PMM+07F91660+07EF1660 CA00
                                                                               


Booting from Hard Disk...
Boot failed: could not read the boot disk

Booting from Floppy...
(QUOTE (1 1))
(1 1)
(QUOTE (1 1))
(1 1)
QEMU: Terminated
```

**Comment by derhexenmeister** [comment]
Here are some observations which are potentially related to the above. One expectation for me is that at least for the C version one would be able to copy and paste lists. But this isn't the case, and there are some other points of confusion for me. For the boot sector version there wouldn't be any way to copy/paste so it's not high priority.

input: (CONS (QUOTE A) (QUOTE B))
output: (A∙B)

; And we can use .
input: (QUOTE (A . B))
output: (A∙B)

; CAR and CDR work for either input format
input: (CAR (CONS (QUOTE A) (QUOTE B)))
output: A

input: (CDR (CONS (QUOTE A) (QUOTE B)))
output: B

input: (CAR (QUOTE (A . B)))
output: A

input: (CDR (QUOTE (A . B)))
output: B

; However A∙B is one symbol, the bullet has no special significance for input
input: (QUOTE (A∙B))
output: (A∙B∙NIL)

input: (CAR (QUOTE (A∙B)))
output: A∙B

input: (CDR (QUOTE (A∙B)))
output: NIL

input: (CAR (QUOTE (A∙B C)))
output: A∙B

input: (CDR (QUOTE (A∙B C)))
output: (C∙NIL)

; Now onto longer lists

; This doesn't work, and outputs two lines
input: (QUOTE ((A . B) (C . D)))
output: ((A∙B)∙NIL)
output: NIL

; We can cons the above into existence though
input: (CONS (CONS (QUOTE A) (QUOTE B)) (CONS (CONS (QUOTE C) (QUOTE D)) NIL))
output: ((A∙B) (C∙D)∙NIL)

; And again - is there an easier method that I'm missing?
input: (CONS (CONS (QUOTE A) (QUOTE B)) (CONS (CONS (QUOTE C) (QUOTE D)) (CONS (CONS (QUOTE E) (QUOTE F)) NIL)))
output: ((A∙B) (C∙D) (E∙F)∙NIL)

**Comment by jart** [comment]
Thanks for the report and investigation. It looks like this was caused by some code cleanup I did before the blog post in 05d2bcbfd95b7afa8f180d528a40d6fbf80875c6 where I forgot to add an explicit NIL compare to `PrintList()`. I've fixed this in 487dc87ab5a575b5c8040b45d030334c1c1a3c0f

**Comment by jart** [comment]
My apologies, correction, this has been fixed in 532b92688e6b1136028ed7a4633811111243592e
